go/vt/topo/stats_conn.go

```diff
@@ -46,21 +53,28 @@ type StatsConn struct {
 	cell     string
 	conn     Conn
 	readOnly bool
+	readSem  *semaphore.Weighted
 }
 
 // NewStatsConn returns a StatsConn
-func NewStatsConn(cell string, conn Conn) *StatsConn {
+func NewStatsConn(cell string, conn Conn, readSem *semaphore.Weighted) *StatsConn {
 	return &StatsConn{
 		cell:     cell,
 		conn:     conn,
 		readOnly: false,
+		readSem:  readSem,
 	}
 }
 
 // ListDir is part of the Conn interface
 func (st *StatsConn) ListDir(ctx context.Context, dirPath string, full bool) ([]DirEntry, error) {
 	startTime := time.Now()
 	statsKey := []string{"ListDir", st.cell}
+	if err := st.readSem.Acquire(ctx, 1); err != nil {
+		return nil, err
+	}
+	defer st.readSem.Release(1)
+	topoStatsConnWaitTimings.Record(statsKey, startTime)
```

Comment:
@deepthi (or anyone) should we reset `startTime` after the semaphore is acquired?

Currently the `topoStatsConnWaitTimings.Record(statsKey, startTime)` will include the time we waited for the semaphore **plus** the operation time

Reply:
Updated to reset `startTime = time.Now()` 👍